build_test/lib/src/resolve_source.dart

```diff
@@ -220,8 +227,11 @@ class _ResolveSourceBuilder<T> implements Builder {
   @override
   Future<void> build(BuildStep buildStep) async {
     if (_resolverFor != buildStep.inputId) return;
-    var result = await _action(buildStep.resolver);
-    onDone.complete(result);
+    try {
+      onDone.complete(await _action(buildStep.resolver));
```

Comment:
[optional] If we aren't worried about `_action` throwing synchronously we can skip the `try/catch`. `Completer.complete` takes a `FutureOr<T>` and forwards async errors correctly.

```dart
onDone.complete(_action(buildStep.resolver));
```

Reply:
We do need to catch synchronous failures as well